File: tests/PraiseBoardTest.php
<?php 
/**
* Praise board tests
*/

require_once 'PraiseBoard.php';

class PraiseBoardTest extends PHPUnit_Framework_TestCase 
{

	var $praiseBoard;

	protected function setUp() 
	{
		$this->praiseBoard = new PraiseBoard;
	}

	protected function tearDown()
	{
		unset($this->praiseBoard);
	}

	public function testEmptyData()
	{
		$this->praiseBoard->setData(array());
		$result = $this->praiseBoard->buildStats();
		$this->assertTrue($result === array());
	}

	public function testOneUserData()
	{
		$data = array('john'=>32);
		$this->praiseBoard->setData($data);
		$result = $this->praiseBoard->buildStats();
		$this->assertEquals($result, $data);
	}

	public function testOrderData()
	{
		$this->praiseBoard->setData(array(
			'john'=>32,
			'mary'=>21,
			'jonathan'=>309,
			'sarah'=>454
		));

		$result = $this->praiseBoard->orderData();

		$this->assertEquals($result, array(
			'sarah'=>454,
			'jonathan'=>309,
			'john'=>32,
			'mary'=>21,
		));
	}

	public function testFindMaxValue()
	{
		$this->praiseBoard->setData(array(
			'vincent'=>20,
			'mary'=>43,
			'caleb'=>29,
		));
		$maxValue = $this->praiseBoard->findMaxValue();
		$this->assertEquals($maxValue, 43);
	}

	public function testBuildStats()
	{
		$this->praiseBoard->setData(array(
			'sarah'=>10,
			'jonathan'=>20,
			'john'=>30,
			'mary'=>5,
		));

		$result = $this->praiseBoard->buildStats();
		$this->assertEquals($result, array(
			'john'=>1,
			'jonathan'=>0.66,
			'sarah'=>0.33,
			'mary'=>0.16
		));

	}
	

}

?>

File: tests/PraiseBoardTest.py
import unittest
from PraiseBoard import PraiseBoard

class PraiseBoardTest(unittest.TestCase):

	def test_empty_input(self):
		praise_board =  PraiseBoard()
		praise_board.setData({})
		self.assertEqual({}, praise_board.build_stats())

	def test_first_rank(self):
		praise_board = PraiseBoard({'mary':23, 'john':100, 'tobby':56})
		most_ranked = praise_board.get_first_rank()
		self.assertEquals(('john', 100), most_ranked)

		

	def test_build_stats(self):
		praise_board = PraiseBoard({'mary':20, 'john':100, 'tobby':10})
		stats = praise_board.build_stats()
		self.assertEquals(stats, [('john',1.0), ('mary',0.2), ('tobby',0.1)])

if __name__ == '__main__':
	unittest.main()
